ecr/fetcher.go

```diff
@@ -143,3 +149,28 @@ func (f *ecrFetcher) doRequest(ctx context.Context, req *http.Request) (*http.Re
 	}
 	return resp, nil
 }
+
+func (f *ecrFetcher) fetchLayerHtcat(ctx context.Context, desc ocispec.Descriptor, downloadURL string) (io.ReadCloser, error) {
+	log.G(ctx).WithField("url", downloadURL).Debug("ecr.fetcher.layer.htcat")
```

Comment:
Manually adding `ecr.fetcher...` prefix looks error-prone. How about either 1) using [SetReportCaller](https://github.com/sirupsen/logrus/blob/de736cf91b921d56253b4010270681d33fdf7cb5/README.md#logging-method-name) or 2) adding a private method on ecrFetcher that returns a logger with a prefix. The prefix can be `WithField("package", "ecr.fetcher...")`

Reply:
`SetReportCaller` adds a fair amount of overhead, so I don't really want to do that.  `WithField("ecr.context", "fetcher.layer.htcat")` looks more reasonable to me, but still ends up being a fair amount of repetition.

I'm not going to change it for this PR, but I've opened https://github.com/awslabs/amazon-ecr-containerd-resolver/issues/6 to follow up later.